packages/fxa-payments-server/src/routes/Checkout/index.tsx

```diff
@@ -176,9 +185,23 @@ export const Checkout = ({
       stripeOverride,
       selectedPlan,
       retryStatus,
+      subscribeToNewsletter,
     ]
   );
 
+  const postSubscriptionAttemptPaypalCallback = useCallback(async () => {
+    if (subscribeToNewsletter) {
+      try {
+        await handleNewsletterSignup();
+      } catch (error) {
+        // TODO: FXA-3667: Show AlertBar on success screen
+        // Note: If both handleNewsletterSignup and fetchProfileAndCustomer fail,
+        // there would be an AlertBar on top of the PaymentErrorView screen.
+      }
+    }
+    await fetchProfileAndCustomer();
```

Comment:
Perhaps do this first, so the customer will see the success screen a little quicker?

Reply:
Good observation; we do call `fetchProfileAndCustomer` (via `onSuccess` callback) first for Stripe.